wcomponents-theme/src/main/js/wc/file/accepted.mjs

```diff
@@ -23,45 +25,64 @@ import getMimeType from "wc/file/getMimeType.mjs";
  * @alias module:wc/file/accepted
  * @requires module:wc/file/getMimeType
  * @param {HTMLInputElement} element A "file input" element.
+ * @param fileInfo The array of files to check, inside an object under "files" property
  */
-export default function accepted(element) {
+export default function accepted(element, fileInfo) {
 	let acceptedType = element.accept;
-	if (acceptedType) {
-		let acceptedTypes = acceptedType.split(",");
-		return getMimeType(element).every((/** @type {module:wc/file/getMimeType~fileType} */fileType) => {
-			let mimeType = fileType.mime || "",
-				extension = fileType.ext ? (`.${fileType.ext}`) : "",
-				passed = (!mimeType && !extension) ||
-						(extension && acceptedType.indexOf(extension) >= 0) ||
-						(mimeType && acceptedType.indexOf(mimeType) >= 0);
-			if (!passed) {  // maybe there is a case difference OR it's a wildcard mimetype or there is some whitespace to be trimmed?
-				for (let i = 0; i < acceptedTypes.length; i++) {
-					let next = acceptedTypes[i].toLowerCase();
-					next = next.trim();
-					if (extension) {
-						extension = extension.toLowerCase();
-						if (extension === next) {
-							passed = true;
-							break;
-						}
-					}
-					if (mimeType) {
-						mimeType = mimeType.toLowerCase();
-						if (mimeType === next) {
-							passed = true;
-							break;
-						} else if (next.indexOf("*") === next.length - 1) {
-							next = next.substring(0, next.length - 1);
-							if (mimeType.indexOf(next) === 0) {
-								passed = true;
-								break;
-							}
-						}
-					}
+
+	if (!acceptedType) return true;
+
+	let acceptedTypes = acceptedType.split(",");
+
+	const checkFileBasic = function (mimeType, extension) {
+		return (!mimeType && !extension) ||
+			(extension && acceptedType.indexOf(extension) >= 0) ||
+			(mimeType && acceptedType.indexOf(mimeType) >= 0);
+	}
+
+	const compareMime = function (actual, reference) {
+		if (actual === reference) {
+			return true;
+		} else if (reference.indexOf("*") === reference.length - 1) {
+			reference = reference.substring(0, reference.length - 1);
+			if (actual.indexOf(reference) === 0) {
+				return true;
+			}
+		}
+		return false;
+	};
+
+	const checkFileArray = function (/** @type {module:wc/file/getMimeType~fileType} */fileType) {
+		let mimeType = fileType.mime || "",
+			extension = fileType.ext ? (`.${fileType.ext}`) : "",
+			passed = checkFileBasic(mimeType, extension);
+
+		if (passed) {
```

Comment:
Do you think this code might be clearer without the `passed` variable? 
Like, we just `return true` instead from all the places, here, line 71 and line 78?
 For example:
```js
if (checkFileBasic(mimeType, extension)) { 
	return true;
}
```
```js
if (extension === next) {
	return true;
}
```

etc